### PR TITLE
bpo-32962: Fix test_gdb failure in debug build with -mcet -fcf-protection -O0

### PR DESCRIPTION
**Pull request by ghost**
This pull request solves problem with reading stack of the interpreter from gdb when new protection flags are used.
The new flags also cause some noise, which is now ignored, on stderr.

https://bugs.python.org/issue32962

### CODE REVIEW

**Comment by vstinner** [comment]
Reminder for reviewers: test_gdb is *not* run on Travis CI. Tests must be run manually before merging such change.

**Comment by stratakis** [comment]
Also in order to test the change, gcc >= 8 is required.

File: Lib/test/test_gdb.py [hunk 1]
```diff
@@ -202,6 +202,8 @@ def get_stack_trace(self, source=None, script=None,
             'BFD: ',
             # ignore all warnings
             'warning: ',
+            # See #32962
+            'Python Exception',
```
Comment (vstinner) [inline Lib/test/test_gdb.py]: Hum, this is not a good idea to ignore bugs :-( Please revert this change.

My https://github.com/python/cpython/pull/7693 should fix this bug.

File: Lib/test/test_gdb.py [hunk 2]
```diff
@@ -162,7 +162,7 @@ def get_stack_trace(self, source=None, script=None,
             commands += ['set print entry-values no']
 
         if cmds_after_breakpoint:
-            commands += cmds_after_breakpoint
+            commands += ['next'] + cmds_after_breakpoint
```
Comment (vstinner) [inline Lib/test/test_gdb.py]: Please add a comment explaining why next is needed. Something like:

```
# bpo-32962: When Python is compiled with -mcet -fcf-protection,
# arguments are unusable before running the first instruction
# of the function entry point.
```

File: Lib/test/test_gdb.py [hunk 3]
```diff
@@ -849,7 +851,7 @@ def __init__(self):
         ''')
         # Verify with "py-bt":
         gdb_output = self.get_stack_trace(cmd,
-                                          cmds_after_breakpoint=['break wrapper_call', 'continue', 'py-bt'])
+                                          cmds_after_breakpoint=['break wrapper_call', 'continue', 'next', 'py-bt'])
```
Comment (vstinner) [inline Lib/test/test_gdb.py]: Ditto: please add a comment.

File: Lib/test/test_gdb.py [hunk 4]
```diff
@@ -161,8 +161,11 @@ def get_stack_trace(self, source=None, script=None,
         if (gdb_major_version, gdb_minor_version) >= (7, 4):
             commands += ['set print entry-values no']
 
+        # bpo-32962: When Python is compiled with -mcet -fcf-protection,
+        # arguments are unusable before running the first instruction
+        # of the function entry point. Next makes the required first step.
```
Comment (vstinner) [inline Lib/test/test_gdb.py]: Would you mind to replace Next with >The 'next' command< to make it more explicit?

Move also the comment inside the if block.

File: Lib/test/test_gdb.py [hunk 5]
```diff
@@ -847,9 +850,12 @@ def __init__(self):
             id("first break point")
             l = MyList()
         ''')
+        # bpo-32962: same case as in get_stack_trace():
+        # we need one extra step in order to read arguments
```
Comment (vstinner) [inline Lib/test/test_gdb.py]: Replace "we need one extra step" with "we need 'next' extra command".

**Review by vstinner — APPROVED** [review]
LGTM, just two minor remarks on comments.

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by vstinner** [comment]
@Traceur759: I'm sorry, I had to revert this change because it caused failures on two buildbots:
https://bugs.python.org/issue32962#msg319669
